src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientsService.kt

```diff
@@ -207,7 +209,8 @@ class ClientsService(
     val clientConfig = clientClientConfigPair.second
     val deployment = getDeployment(clientId)
     setValidDays(clientConfig)
-    return ClientComposite(client, clientConfig, retrieveAuthorizationConsent(client), deployment)
+
+    return ClientComposite(client, clientConfig, retrieveAuthorizationConsent(client), deployment, authService.getServiceRoles(clientIdService.toBase(clientId)))
```

Comment:
It is only authorisation code grant type clients that have service roles so we should suppress this call for client credentials grant type clients

Reply:
Updated PR